kotlin/export-tiles/src/main/java/com/example/exporttiles/MainActivity.kt

```diff
@@ -0,0 +1,240 @@
+/*
+ * Copyright 2020 Esri
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package com.example.exporttiles
+
+import android.content.DialogInterface
+import android.graphics.Color
+import android.graphics.Point
+import android.graphics.PorterDuff
+import android.graphics.PorterDuffColorFilter
+import android.os.Bundle
+import android.util.Log
+import android.view.View
+import android.widget.Toast
+import androidx.appcompat.app.AlertDialog
+import androidx.appcompat.app.AppCompatActivity
+import androidx.core.content.ContextCompat
+import com.esri.arcgisruntime.concurrent.Job
+import com.esri.arcgisruntime.concurrent.ListenableFuture
+import com.esri.arcgisruntime.data.TileCache
+import com.esri.arcgisruntime.geometry.Envelope
+import com.esri.arcgisruntime.layers.ArcGISTiledLayer
+import com.esri.arcgisruntime.loadable.LoadStatus
+import com.esri.arcgisruntime.mapping.ArcGISMap
+import com.esri.arcgisruntime.mapping.Basemap
+import com.esri.arcgisruntime.mapping.Viewpoint
+import com.esri.arcgisruntime.mapping.view.Graphic
+import com.esri.arcgisruntime.mapping.view.GraphicsOverlay
+import com.esri.arcgisruntime.symbology.SimpleLineSymbol
+import com.esri.arcgisruntime.tasks.tilecache.ExportTileCacheJob
+import com.esri.arcgisruntime.tasks.tilecache.ExportTileCacheParameters
+import com.esri.arcgisruntime.tasks.tilecache.ExportTileCacheTask
+import kotlinx.android.synthetic.main.activity_main.*
+import kotlinx.android.synthetic.main.dialog_layout.*
+import java.io.File
+
+class MainActivity : AppCompatActivity() {
+
+  private val TAG: String = MainActivity::class.java.simpleName
+
+  override fun onCreate(savedInstanceState: Bundle?) {
+    super.onCreate(savedInstanceState)
+    setContentView(R.layout.activity_main)
+
+    // create an ArcGISTiledLayer to use as the basemap
+    val tiledLayer = ArcGISTiledLayer(getString(R.string.world_street_map))
+    val map = ArcGISMap().apply {
+      basemap = Basemap(tiledLayer)
+      minScale = 10000000.0
+    }
+    // create a graphics overlay
+    val graphicsOverlay = GraphicsOverlay()
```

Comment:
Could reorder a few of these instantiations to use all the applyz

```
    // create an ArcGISTiledLayer to use as the basemap
    val tiledLayer = ArcGISTiledLayer(getString(R.string.world_street_map))
    val map = ArcGISMap().apply {
      basemap = Basemap(tiledLayer)
      minScale = 10000000.0
    }

    // create a graphic to show a red box around the tiles we want to download
    val downloadArea = Graphic().apply {
      symbol = SimpleLineSymbol(SimpleLineSymbol.Style.SOLID, Color.RED, 2f)
    }
    
    // create a graphics overlay
    val graphicsOverlay = GraphicsOverlay().apply { 
      graphics.add(downloadArea)
    }
    
    mapView.apply {
      // set the map to the map view
      this.map = map
      setViewpoint(Viewpoint(51.5, 0.0, 10000000.0))
      // add the graphics overlay to the map view
      graphicsOverlays.add(graphicsOverlay)
    }
```

Reply:
Haha so many applyz! I'm hesitant to use apply for single line instances, purely because it actually adds an extra line of code doing it this way and it doesn't make it quite so readable e.g. the above would mean 6 lines of codes, whereas I think the following is probably more readable and is only 4 lines: 

```
    val graphicsOverlay = GraphicsOverlay()
    val downloadArea = Graphic()
    graphicsOverlay.graphics.add(downloadArea)
    downloadArea.symbol = SimpleLineSymbol(SimpleLineSymbol.Style.SOLID, Color.RED, 2f)
```

Reply:
Hmm...I find the 6 lines more readable myself (I think the indents help). It's more lines, but arguably less code. But anyways, if you're keen on the more java-y build pattern I'd at least suggest re-ordering it like:

```
    val downloadArea = Graphic()
    downloadArea.symbol = SimpleLineSymbol(SimpleLineSymbol.Style.SOLID, Color.RED, 2f)

    val graphicsOverlay = GraphicsOverlay()
    graphicsOverlay.graphics.add(downloadArea)
```